spoilerLog.py

```diff
@@ -0,0 +1,112 @@
+import json
+import logic
+import explorer
+import patches.witch
+import patches.titleScreen
+import patches.dungeonEntrances
+from locations.witch import Witch
+
+class RaceRomException(Exception):
+    pass
+
+class SpoilerItemInfo():
+    def __init__(self, ii, rom, multiworld):
+        self.id = ii.nameId
+        self.area = ii.metadata.area
+        self.locationName = ii.metadata.name
+        self.itemName = str(ii.read(rom))
+        self.player = None
+
+        if multiworld:
+            self.player = rom.banks[0x3E][0x3300 + ii.room] + 1
+    
+    def __repr__(self):
+        itemName = self.itemName
+
+        if self.player:
+            itemName = "P%s's %s" % (self.player, itemName)
+
+        return "%25s at %s - %s" % (itemName, self.area, self.locationName) 
+
+class SpoilerLog():
+    def __init__(self, args, rom):
+        if rom.banks[0][7] == 0x01:
+            raise RaceRomException()
+
+        self.seed = patches.titleScreen.readSeed(rom)
+        self.testOnly = args.test
+        self.accessibleItems = []
+        self.inaccessibleItems = None
+        self.dungeonOrder = patches.dungeonEntrances.readEntrances(rom)
+        self.outputFormat = args.spoilerformat
+
+        # Assume the broadest settings if we're dumping a seed we didn't just create
+        if args.dump:
+            # The witch flag causes trouble if we blindly turn it on
```

Comment:
What kind of problems does it cause? It should just have a toadstool and magic powder logic wise...

Reply:
I didn't look into the root cause, but I'd get logic failures really early, often leaving the vast majority of items "inaccessible".  I'd guess it's related to the toadstool/powder split, but I'm not sure.

Reply:
Mmm, I'll have a look at it. Should be something simple to fix